src/ORM/DataList.php

```diff
@@ -943,10 +943,19 @@ public function setQueriedColumns($queriedColumns)
      * Filter this list to only contain the given Primary IDs
      *
      * @param array $ids Array of integers
+     * @throws InvalidArgumentException
      * @return $this
      */
     public function byIDs($ids)
     {
+        foreach ($ids AS $id) {
+            if (!is_int($id)) {
```

Comment:
You can check that the card value is the same as the original.

```php
If (is_numeric($val) && intval($val) == $val)
```

This should do strings getting through as ints and floats being cast to int.